cli/flags.rs

```diff
@@ -1323,7 +1324,7 @@ fn resolve_hosts(paths: Vec<String>) -> Vec<String> {
     let parts = host_and_port.split(':').collect::<Vec<&str>>();
```

Comment:
I think this is more of the issue, naively splitting on `:`

Looks like resolve_hosts could be updated to support bracketed based host names (https://tools.ietf.org/html/rfc2732.html) (and then parsing IP based hostnames?)